lib/improver/tests/utilities/test_statistical_operations.py

```diff
@@ -0,0 +1,388 @@
+        expected = set_reference_probabilities()
+
+        probability_cube = ProbabilitiesFromPercentiles2D(
+            percentiles_cube).percentile_interpolation(
+                self.orography_cube, percentiles_cube)
+        self.assertArrayAlmostEqual(probability_cube.data, expected)
+
+    def test_values_inverse_ordering(self):
+        """
+        Test that interpolated probabilities at given topography heights are
+        sensible when we use the inverse_ordering set to True. This is for
+        situations in which the values associated with the percentiles increase
+        in the opposite direction, e.g. 0 % = 100m, 20% = 50m, etc.
+
+        In this situation we expect the lowest points to have a probability of
+        1, and the highest points to have probabilities of 0. The probabilities
+        between should be the inverse of what is found in the usual case.
+        """
+        percentiles_cube = set_up_percentiles_cube()
+        # Invert the values associated with the percentiles.
+        percentiles_cube.data = np.flipud(percentiles_cube.data)
+        expected = set_reference_probabilities()
+        expected = 1.0 - expected
+
+        probability_cube = ProbabilitiesFromPercentiles2D(
+            percentiles_cube).percentile_interpolation(
+                self.orography_cube, percentiles_cube)
+        self.assertArrayAlmostEqual(probability_cube.data, expected)
+
+    def test_equal_percentiles(self):
+        """
+        Test for sensible behaviour when some percentile levels are equal
+        """
+        percentiles_cube = set_up_percentiles_cube()
+        percentiles_cube.data[0].fill(300.)
+        expected = set_reference_probabilities()
+        expected[np.where(expected < 0.25)] = 0.
+        probability_cube = ProbabilitiesFromPercentiles2D(
+            percentiles_cube).percentile_interpolation(
+                self.orography_cube, percentiles_cube)
+
+        self.assertArrayAlmostEqual(probability_cube.data, expected)
+
+    def test_equal_percentiles_inverse_ordering(self):
+        """
+        Test for sensible behaviour when some percentile levels are equal
+        in the case of inverse ordering (as described above).
+        """
+        percentiles_cube = set_up_percentiles_cube()
+        percentiles_cube.data[0].fill(300.)
+        # Invert the values associated with the percentiles.
+        percentiles_cube.data = np.flipud(percentiles_cube.data)
+        expected = set_reference_probabilities()
+        expected[np.where(expected <= 0.25)] = 0.
+        expected = 1.0 - expected
+        probability_cube = ProbabilitiesFromPercentiles2D(
+            percentiles_cube).percentile_interpolation(
+                self.orography_cube, percentiles_cube)
+        self.assertArrayAlmostEqual(probability_cube.data, expected)
+
+
+class Test_process(IrisTest):
+    """
+    Test top level processing function that calls percentile_interpolation()
+    """
```

Comment:
Rethinking ... I think that throughout this test class we should be comparing the shape of the probability cube with the percentiles cube rather than the orography (threshold) cube, to account for any extra dimensions (as eg at line 335).  Changes needed at lines 289, 299 and 381.

Reply:
I've created a reference_cube to use based upon one slice of the percentiles_cube; we obviously have to dispose of the percentiles dimension.